src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationServiceTest.java

```diff
@@ -386,11 +393,61 @@ void validateCurrentPositionCaseClosed(String state, boolean expected) {
     void shouldValidateJurisdictionCodesWithEmptyCodes() {
         List<String> errors = new ArrayList<>();
         caseDetails3.getCaseData().setJudgementCollection(new ArrayList<>());
-        eventValidationService.validateJurisdictionCodes(caseDetails3.getCaseData(), errors);
+        eventValidationService.validateJurisdiction(caseDetails3.getCaseData(), errors);
 
         assertEquals(0, errors.size());
     }
 
+    @Test
+    void shouldValidateDisposalDateInFuture() {
+        List<String> errors = new ArrayList<>();
+        eventValidationService.validateJurisdiction(setCaseDataForDisposalDateTest("2777-12-23","2022-01-20T00:00:00.000"), errors);
+        assertEquals(EventValidationService.DISPOSAL_DATE_IN_FUTURE, errors.get(0));
+    }
+
+    private HearingTypeItem setHearing(String hearingDate) {
+        HearingTypeItem hearingTypeItem = new HearingTypeItem();
+        hearingTypeItem.setId(UUID.randomUUID().toString());
+        HearingType hearingType = new HearingType();
+        DateListedTypeItem dateListedTypeItem = new DateListedTypeItem();
+        DateListedType dateListedType = new DateListedType();
+        dateListedType.setListedDate(hearingDate);
+        dateListedTypeItem.setId(UUID.randomUUID().toString());
+        dateListedTypeItem.setValue(dateListedType);
+        hearingType.setHearingDateCollection(Collections.singletonList(dateListedTypeItem));
+        hearingTypeItem.setValue(hearingType);
+        return hearingTypeItem;
+    }
+
+    private CaseData setCaseDataForDisposalDateTest(String disposalDate, String hearingDate) {
+        CaseData caseData = new CaseData();
+        HearingTypeItem hearingTypeItem1 = setHearing(hearingDate);
+        HearingTypeItem hearingTypeItem2 = setHearing("2021-01-01T00:00:00.000");
+        caseData.setHearingCollection(Arrays.asList(hearingTypeItem1, hearingTypeItem2));
+        JurCodesTypeItem jurCodesTypeItem = new JurCodesTypeItem();
+        jurCodesTypeItem.setId(UUID.randomUUID().toString());
+        JurCodesType jurCodesType = new JurCodesType();
+        jurCodesType.setJuridictionCodesList("blah blah");
+        jurCodesType.setDisposalDate(disposalDate);
+        jurCodesTypeItem.setValue(jurCodesType);
+        caseData.setJurCodesCollection(Collections.singletonList(jurCodesTypeItem));
+        return caseData;
+    }
+
+    @Test
+    void disposalDateNoMatchWithHearingDate() {
+        List<String> errors = new ArrayList<>();
+        eventValidationService.validateJurisdiction(setCaseDataForDisposalDateTest("2022-06-30", "2020-05-01T10:10:00.000"), errors);
+        assertEquals(EventValidationService.DISPOSAL_DATE_HEARING_DATE_MATCH, errors.get(0));
+    }
+
+    @Test
+    void disposalDateMatchWithHearingDate() {
+        List<String> errors = new ArrayList<>();
+        eventValidationService.validateJurisdiction(setCaseDataForDisposalDateTest("2022-06-30", "2022-06-30T00:00:00.000"), errors);
```

Comment:
Date strings would be better held as constants

Reply:
using dates as constants now